src/features/modals/CreateProjectModal.tsx

```diff
@@ -1,59 +1,57 @@
-import { signal } from "@preact/signals-core";
-import { type Ref, useRef, useState } from "preact/hooks";
+import { useRef, useState } from "preact/hooks";
 import { PlusIcon } from "../icons/PlusIcon";
 import { create, useProjectsList } from "../../lib/projects";
+import { Modal } from "./Modal";
 
-const refSignal = signal<Ref<HTMLDialogElement> | null>(null);
```

Comment:
All this signal stuff was overkill. Streamlined it with internal refs and keeping the modal within the modal button component.